src/App.js

```diff
@@ -1,101 +1,129 @@
+import './App.css'
+
+// Defining initial state outside of the component can make the component easier to read.
+const timersInitialState = [
+  {
+    id: 1,
+    name: 'Exercise 1',
+    seconds: 20,
+    minutes: 1,
+    // I'm not going to alter the shape of this object since it is used all over the app,
+    //but I would suggest not storing anything in state that can be computed from other values in the state.
+    // For example, `isComplete`, can be calculated on the fly from `currentRep` and `targetRep`. (const isComplete = currentRep >= targetRep)
+    isComplete: false,
+    currentRep: 0,
+    targetRep: 1,
+  },
+  {
+    id: 2,
+    name: 'Exercise 2',
+    seconds: 30,
+    minutes: 0,
+    isComplete: false,
+    currentRep: 0,
+    targetRep: 2,
+  },
+  {
+    id: 3,
+    name: 'Exercise 3',
+    seconds: 5,
+    minutes: 0,
+    isComplete: false,
+    currentRep: 0,
+    targetRep: 1,
+  },
+]
 
 function App() {
-  const [timers, setTimers] = useState([
-      {
-        id: 1,
-        name: "Exercise 1",
-        seconds: 20,
-        minutes: 1,
-        isComplete: false,
-        currentRep: 0,
-        targetRep: 1,
-      },
+  const [timers, setTimers] = useState(timersInitialState)
+
+  function changeCompletion_old(id) {
+    // This function is very hard to read & follow what it is doing. Good variable naming and Array methods can help improve it.
+    // I am leaving it here for reference, but renamed it to `changeCompletion_old`
+    setTimers((prevTimer) => [
+      ...prevTimer.slice(0, id - 1), // making assumptions about how the `id` relates to a timer's index is fragile and bug prone. Use the id to find the index using prevTimer.findIndex(timer => timer.id === id)
       {
-        id: 2,
-        name: "Exercise 2",
-        seconds: 30,
-        minutes: 0,
-        isComplete: false,
-        currentRep: 0,
-        targetRep: 2
+        ...prevTimer[id - 1],
+        isComplete: prevTimer[id - 1].currentRep >= prevTimer[id - 1].targetRep - 1 && true, // it works due to - 1, but i think there's something wrong with useEffect
+        currentRep: prevTimer[id - 1].currentRep + 1,
       },
-      {
-        id: 3,
-        name: "Exercise 3",
-        seconds: 5,
-        minutes: 0,
-        isComplete: false,
-        currentRep: 0,
-        targetRep: 1,
-      }
-  ])
+      ...prevTimer.slice(id),
+    ])
+  }
 
   function changeCompletion(id) {
-    setTimers(prevTimer => (
-      [
-        ...prevTimer.slice(0,id - 1),
-        {
-          ...prevTimer[id - 1],
-          isComplete: prevTimer[id - 1].currentRep >= prevTimer[id - 1].targetRep - 1 && true,  // it works due to - 1, but i think there's something wrong with useEffect
-          currentRep: prevTimer[id - 1].currentRep + 1
-        },
-        ...prevTimer.slice(id)
-    ]))
+    const newTimers = timers.map((timer) => ({ ...timer })) // make a deep copy of an array of objects so you don't accidentally update state directly
```

Comment:
I think @nileshlaxmi meant to say `timers?.map`, which would check that timers has a map method before calling it, which is good practice if you can't rely on whether or not it would be there (like if `timers` was null for some reason).